presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkConfig.java

```diff
@@ -477,4 +499,38 @@ public PrestoSparkConfig setAdaptiveJoinSideSwitchingEnabled(boolean adaptiveJoi
         this.adaptiveJoinSideSwitchingEnabled = adaptiveJoinSideSwitchingEnabled;
         return this;
     }
+
+    /**
+     * Parses the catalog properties from the input string in format catalog.key=value and returns a nested map of properties.
+     *
+     * @throws IllegalArgumentException If the input string or any of its properties are not correctly formatted.
+     */
+    public static Map<String, Map<String, String>> parseCatalogProperties(String input)
+            throws IllegalArgumentException
+    {
+        try {
+            Map<String, String> inputMap = MAP_SPLITTER.split(nullToEmpty(input));
+
+            Map<String, Map<String, String>> result = new HashMap<>();
+
+            for (Map.Entry<String, String> entry : inputMap.entrySet()) {
+                String key = entry.getKey();
+                String value = entry.getValue();
+
+                String[] parts = key.split("\\.");
+                if (parts.length != 2) {
+                    throw new IllegalArgumentException("Property " + key + " is not correctly formatted");
+                }
+                String catalogName = parts[0];
+                String propertyName = parts[1];
```

Comment:
I don't think this is correct.
1. the property name will include the catalogName part, e.g. the name of the property is `hive.ignore-table-bucketing`, not just `ignore-table-bucketing`
2. The name of the catalog isn't necessarily the same as the prefix to the property.  by convention catalog properties are prefixed with the connector name, but that's not a requirement, and also the connector name is not the same as the catalog name (a catalog might be said to be an instance of a specific connector). e.g. we call our hive catalog extension prism, but inherit hive config properties, and the config property name would still be `hive.ignore-table-bucketing`. similarly you can have two catalogs from the same connector (e.g. prism and prism-batch), both of which can set the `hive.ignore-table-bucketing` property. I think we need a different format for the input here that explicitly maps properties by catalog name.  I would also test what happens if you add properties for a catalog that isn't registered.

Reply:
I believe in this case the name of the catalog session property is just `ignore_table_bucketing`

https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/HiveSessionProperties.java#L50

So, if the config string is in format `catalog.ignore_table_bucketing=true`, it should be parsed correctly.

Reply:
discussed offline, and I was thinking these were going to be a list of config properties, but actually it's a list of session properties.  In that case this change should work fine.  

we do have some similar code, but not enough that it's especially important to share between the two: https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/server/HttpRequestSessionContext.java#L189-L198.  But you can use similar to https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/server/HttpRequestSessionContext.java#L204-L205 to create an immutableMap at the end.

However, we should make clear in the name that this is specifically for session properties.  Probably even instead of having a separate property for this, just add this functionality to support catalog properties to existing session properties retry property.  You can see how the HttpRequestSessionContext handles both at the same time.

Reply:
Adjusted according to your suggestion. `spark.retry-presto-session-properties` now supports both session and catalog properties.